File: 0801_0900/0822/0822.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    x1, y1, x2, y2, x3, y3 = (float(x) for x in f.read().split())
    q.write(str(abs(((x2 - x1) * (y3 - y1) - (y2 - y1) * (x3 - x1)) / 2)))

File: 0801_0900/0822/0822.c
#include <stdio.h>
#include <math.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	double x1, y1, x2, y2, x3, y3;

	fscanf(f, "%lf%lf%lf%lf%lf%lf", &x1, &y1, &x2, &y2, &x3, &y3);
	fprintf(q, "%f", fabs(((x2 - x1) * (y3 - y1) - (y2 - y1) * (x3 - x1)) / 2));
	return 0;
}
